packages/graph/hash_graph/lib/graph/src/types/schema/helper.rs

```diff
@@ -0,0 +1,291 @@
+        let one_of = Self::new_unchecked(one_of);
+        one_of.validate()?;
+        Ok(one_of)
+    }
+
+    #[must_use]
+    pub fn one_of(&self) -> &[T] {
+        &self.one_of
+    }
+}
+
+impl<T> Validate for OneOf<T> {
+    fn validate(&self) -> Result<(), ValidationError> {
+        if self.one_of.is_empty() {
+            return Err(ValidationError::OneOfEmpty);
+        }
+        Ok(())
+    }
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OneOfRepr<T> {
+    one_of: Vec<T>,
+}
+
+impl<T> TryFrom<OneOfRepr<T>> for OneOf<T> {
+    type Error = ValidationError;
+
+    fn try_from(one_of: OneOfRepr<T>) -> Result<Self, Self::Error> {
+        Self::new(one_of.one_of)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::error::Error;
+
+    use serde_json::json;
+
+    use super::*;
+    use crate::types::schema::Uri;
+
+    #[test]
+    fn array() -> Result<(), Box<dyn Error>> {
+        let array = Array::new(Uri::new("https://example.com/data_type")?, None, None)?;
+
+        let json = serde_json::to_value(&array)?;
+        assert_eq!(
+            json,
+            serde_json::json!({
+                "type": "array",
+                "items": "https://example.com/data_type",
+            })
+        );
+
+        let array2 = serde_json::from_value(json)?;
+        assert_eq!(array, array2);
+
+        let array = Array::new(Uri::new("https://example.com/data_type")?, 1, 2)?;
+
+        let json = serde_json::to_value(&array)?;
+        assert_eq!(
+            json,
+            serde_json::json!({
+                "type": "array",
+                "items": "https://example.com/data_type",
+                "minItems": 1,
+                "maxItems": 2,
+            })
+        );
+
+        let array2 = serde_json::from_value(json)?;
+        assert_eq!(array, array2);
+
+        let json = json!({
+            "items": "https://example.com/data_type"
+        });
+        assert!(serde_json::from_value::<Array<Uri>>(json).is_err());
+
+        let json = json!({
+            "type": "array",
+            "items": "https://example.com/data_type",
+            "minItems": 1
+        });
+        assert!(serde_json::from_value::<Array<Uri>>(json).is_ok());
+
+        let json = json!({
+            "type": "array",
+            "items": "https://example.com/data_type",
+            "maxItems": 1
+        });
+        assert!(serde_json::from_value::<Array<Uri>>(json).is_ok());
+
+        let json = json!({
+            "type": "array",
+            "minItems": 2,
+            "maxItems": 1
+        });
+        assert!(serde_json::from_value::<Array<Uri>>(json).is_err());
+
+        let json = json!({
+            "type": "array",
+            "items": "https://example.com/data_type",
+            "numItems": 1,
+        });
+        assert!(serde_json::from_value::<Array<Uri>>(json).is_err());
+
+        Ok(())
+    }
+
+    #[test]
+    fn one_or_many_single() -> Result<(), Box<dyn Error>> {
```

Comment:
This and the `one_or_many_array` test below still uses the old names :)

Reply:
Addressed in c7d4f6c53194bf4adcf70b3c376160614371b5d0